assets/js/modules/analytics/components/common/AcquisitionPieChart.js

```diff
@@ -67,14 +66,10 @@ const GOOGLE_CHART_PIE_SETTINGS = {
 };
 
 function AcquisitionPieChart( { data, args, source } ) {
-	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
-	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
-	const internalWebPropertyID = useSelect( ( select ) => select( STORE_NAME ).getInternalWebPropertyID() );
 	const url = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityURL() );
-	const sourceURI = useSelect( ( select ) => select( STORE_NAME ).getServiceURL(
-		{
-			path: applyEntityToReportPath( url, `/report/trafficsources-overview/a${ accountID }w${ internalWebPropertyID }p${ profileID }/` ),
-		} ) );
+	const sourceURI = useSelect( ( select ) => select( STORE_NAME ).getServiceReportURL( 'trafficsources-overview', {
+		'_r.drilldown': `analytics.pagePath:${ url }`,
+	} ) );
```

Comment:
You unconditionally add this `url`-based report argument here, which doesn't cater for the fact that `url` can be `null`. `url` is only not null on the single URL details view. `applyEntityToReportPath` had a conditional for this, and we need to maintain this here. Right now you'll see a bunch of links including `null` for that reason.